proposals/0000-ios-scenedelegate-and-resizing.md

```diff
@@ -0,0 +1,173 @@
+---
+title: iOS Migration to SceneDelegate
+author:
+- artus9033
+date: 16.12.2025
+---
+
+# RFC0000: [iOS] Migration to `SceneDelegate`
+
+## Summary
+
+iOS26 introduced deprecation of many UIApplication APIs and made `SceneDelegate` API the preferred one, notifying programmers with a warning that "UIScene lifecycle will soon be required". In future versions of iOS, `SceneDelegate` [is going to be the only supported API](https://developer.apple.com/documentation/technotes/tn3187-migrating-to-the-uikit-scene-based-life-cycle#:~:text=Failure%20to%20adopt%20will%20result%20in%20an%20assert%20in%20the%20future.) and therefore we need to start migrating to it.
+
+![](assets/0000-ios-deprecation-warning.png)
+
+It is possible to perform dynamic window resizing both via stage manager (split view) and freely via the bottom-right corner handle. `SceneDelegate` with proper configuration in `Info.plist` can enable the multi-window capability for iPadOS, which would require adjustments in RN code and RN libraries code to accommodate such design. Such a change would be large and shall be addressed in a separate RFC. The scope of this RFC is to cover a single resizable window (single instance of a given RN app), yielding the assumption that `UIApplicationSupportsMultipleScenes` **must not** be set to true at the moment. This RFC will bring in an additive change that should allow for adoption of a new `SceneDelegate` entrypoint while also allowing for (deprecated) usage of `AppDelegate`. Also, the `@react-native-community/template` and `packages/rn-tester` should be migrated to implement the `SceneDelegate` API.
+
+One related aspect not covered by this RFC is the performance of `useWindowDimensions` hook, which during intensive resizing of the application behaves suboptimal. This matter shall be addressed in a separate RFC.
+
+The idea proposed in this RFC is to:
+
+1) migrate from `AppDelegate` to `SceneDelegate`  
+2) educate on the need to migrate existing apps to adopt `SceneDelegate`
```

Comment:
```suggestion
```

This can be removed. 1) is the topic of the RFC, and 2) is implicit: part of an RFC is educating the users.